trie/trie.go

```diff
@@ -47,6 +48,10 @@ type LeafCallback func(leaf []byte, parent common.Hash) error
 type Trie struct {
 	db   *Database
 	root node
+	// Keep a rough track of the number of leafs to commit
+	dirtyCount int
```

Comment:
Might also add to the docs that the statedb deduplicates writes and only does a trie update when the data is actually pushed to the database, so this counter will actually be pretty precise.

Reply:
It will be _pretty_ precise, but I'm increasing it for deletions aswell, so that will cause some skew

Reply:
We should probably drop this field if it's not used (yet)